test/ChainlinkOracleTest.sol

```diff
@@ -133,15 +143,48 @@ contract ChainlinkOracleTest is Test {
         assertApproxEqRel(oracle.price(), expectedPrice, deviation);
     }
 
+    function testEthSDaiOracle() public {
+        ChainlinkOracle oracle =
+            new ChainlinkOracle(vaultZero, 1, sDaiVault, 1e18, feedZero, feedZero, daiEthFeed, feedZero, 18, 18);
+        (, int256 quoteAnswer,,,) = daiEthFeed.latestRoundData();
+        assertEq(
+            oracle.price(),
+            // 1e(36 + dQ1 + fpQ1 + fpQ2 - dB1 - fpB1 - fpB2) * qCS / bCS
+            10 ** (36 + 18 + 18 + 0 - 18 - 0 - 0) * 1e18 / (sDaiVault.convertToAssets(1e18) * uint256(quoteAnswer))
+        );
+    }
+
+    function testUsdcSDaiOracle() public {
+        ChainlinkOracle oracle =
+            new ChainlinkOracle(vaultZero, 1, sDaiVault, 1e18, usdcEthFeed, feedZero, daiEthFeed, feedZero, 6, 18);
+        (, int256 baseAnswer,,,) = usdcEthFeed.latestRoundData();
+        (, int256 quoteAnswer,,,) = daiEthFeed.latestRoundData();
+        // 1e(36 + dQ1 + fpQ1 + fpQ2 - dB1 - fpB1 - fpB2) * qCS / bCS
+        uint256 scaleFactor = 10 ** (36 + 18 + 18 + 0 - 6 - 18 - 0) * 1e18;
+        assertEq(
+            oracle.price(),
+            scaleFactor.mulDiv(uint256(baseAnswer), (sDaiVault.convertToAssets(1e18) * uint256(quoteAnswer)))
+        );
+        // DAI has 12 more decimals than USDC.
+        uint256 expectedPrice = 10 ** (36 + 12);
+        // Admit a 50% interest gain before breaking this test.
+        uint256 deviation = 0.66 ether;
```

Comment:
I think this corresponds to 200% interest gain. Take `a = oracle.price(), b = expectedPrice`, and say `b = 1`, then `assertApproxEqRel` computes `(b - a) / b`. So this is equal to `0.66` when `a` is 0.33, which corresponds to the price having tripled. I think `deviation = 0.33 ether` gives 50%